src/transformStateWithClones.js

```diff
@@ -8,6 +8,32 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+
+  let copy = { ...state };
+  const story = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        copy = {};
+        break;
+
+      case 'addProperties':
+        copy = { ...copy, ...action.extraData };
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete copy[key];
+        });
+        break;
+
+      default:
```

Comment:
[CODE STYLE]: The `switch` statement has a `default` case for error handling, which is good practice.